src/data_distribution/global_ptr.hpp

```diff
@@ -101,19 +101,15 @@ namespace argo {
 				 * @brief return the home node of the value pointed to, or a
 				 * default value if the page has not yet been first-touched
 				 * under first-touch allocation.
-				 * @return home node id
+				 * @return home node id, or argo::data_distribution::invalid_node_id
+				 * if access_ptr has not been first-touched yet
 				 */
 				node_id_t peek_node() {
 					// If homenode is not yet calculated we need to find it
 					if(homenode == invalid_node_id) {
-						// Do not invoke first-touch
-						if(is_first_touch_policy()) {
-							homenode = policy()->peek_homenode(
-									reinterpret_cast<char*>(access_ptr));
-						} else {
-							homenode = policy()->homenode(
-									reinterpret_cast<char*>(access_ptr));
-						}
+						// Avoid invoking first-touch
+						homenode = policy()->peek_homenode(
+								reinterpret_cast<char*>(access_ptr));
```

Comment:
should this avoid the write to `homenode` iff there is no change, or is that premature optimization?

Reply:
Since I corrected the comment above I applied changes along these lines as well.